### PR TITLE
Added support for watchOS 2

### PR DESCRIPTION
**Pull request by Fudge0952**
Added support for watchOS 2

### CODE REVIEW

File: PINCache/PINMemoryCache.m [hunk 1]
```diff
@@ -79,7 +79,7 @@ - (instancetype)init
         _removeAllObjectsOnMemoryWarning = YES;
         _removeAllObjectsOnEnteringBackground = YES;
 
-        #if __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_4_0
+		#if TARGET_OS_IPHONE && defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && !TARGET_OS_WATCH
```
Comment (garrettmoon) [inline PINCache/PINMemoryCache.m]: Does this not still need a check to make sure __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_4_0?

Comment (Fudge0952) [inline PINCache/PINMemoryCache.m]: Hmm must of slipped out of the code when I was adding watch support. I have updated the PR

File: PINCache/PINMemoryCache.m [hunk 2]
```diff
@@ -111,7 +111,7 @@ + (instancetype)sharedCache
 
 - (void)didObserveApocalypticNotification:(NSNotification *)notification
 {
-    #if __IPHONE_OS_VERSION_MIN_REQUIRED >= __IPHONE_4_0
+    #if TARGET_OS_IPHONE && defined(__IPHONE_OS_VERSION_MIN_REQUIRED) && !TARGET_OS_WATCH
```
Comment (garrettmoon) [inline PINCache/PINMemoryCache.m]: Oh! Here too :)

Comment (Fudge0952) [inline PINCache/PINMemoryCache.m]: >_< good spot. I searched and found no more. Updated again :)

**Pull request merged**